helix-core/src/main/java/org/apache/helix/manager/zk/ParticipantManager.java

```diff
@@ -339,21 +361,24 @@ private void createLiveInstance() {
    * carry over current-states from last sessions
    * set to initial state for current session only when state doesn't exist in current session
    */
-  private void carryOverPreviousCurrentState() {
-    List<String> sessions = _dataAccessor.getChildNames(_keyBuilder.sessions(_instanceName));
+  public static synchronized void carryOverPreviousCurrentState(HelixDataAccessor dataAccessor,
+      PropertyKey.Builder keyBuilder, String instanceName, String sessionId,
```

Comment:
You can remove the keyBuilder here. Because it can be fetched as:

accessor.keyBuilder().

So there are several places you can simplify the code.